apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/GraphQLCompiler.kt

```diff
@@ -26,8 +26,9 @@ class GraphQLCompiler {
         generateVisitorForPolymorphicDatatypes = args.generateVisitorForPolymorphicDatatypes
     )
 
-    if (args.packageNameProvider.schemaPackageName.isNotBlank()) {
-      File(args.outputDir, args.packageNameProvider.schemaPackageName.replace('.', File.separatorChar)).deleteRecursively()
+    val schemaPackageName = (args.packageNameProvider as? DeprecatedPackageNameProvider)?.schemaPackageName
```

Comment:
Just a question. If this is not the `Deprecated` one, the operation gets ignored. Is that expected?

Reply:
This deletes the output directory (I guess to prevent stale files). But this is already done there: https://github.com/apollographql/apollo-android/blob/e1084c9d9e6a47fc1c12a230c5c543e9cd2d9368/apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloGenerateSourcesTask.kt#L77

So it's no use doing it twice. I just didn't touch the current behaviour for the deprecated plugin to be on the safe side.